types/vote.go

```diff
@@ -226,6 +157,9 @@ func (vote *Vote) Verify(chainID string, pubKey crypto.PubKey) error {
 	if !pubKey.VerifySignature(VoteSignBytes(chainID, v), vote.Signature) {
 		return ErrVoteInvalidSignature
 	}
+	if vote.Extension != nil && !pubKey.VerifySignature(vote.Extension, vote.ExtensionSignature) {
```

Comment:
You probably need to pass a canonical extension structure here (marshalled) rather that the extension's bytes directly

Reply:
Done